test/jdk/javax/imageio/spi/AppletContextTest/IIOPluginTest.java

```diff
@@ -22,22 +22,24 @@
  */
 
 import javax.imageio.ImageIO;
+import java.util.ServiceConfigurationError;
 
 public class IIOPluginTest {
 
     public static String[] dummyformatNames = {"test_5076692", "TEST_5076692"};
     public static String[] dummymimeType = {"image/test_5076692"};
 
     public static void main(String[] args) {
-        SecurityManager sm = System.getSecurityManager();
-        System.out.println("Sm is " + sm);
-
-        String formatNames[] = ImageIO.getReaderFormatNames();
-        String readerMimeTypes[] = ImageIO.getReaderMIMETypes();
-
-        if (!isPresent(dummyformatNames, formatNames) ||
-            !isPresent(dummymimeType, readerMimeTypes)) {
-            throw new RuntimeException("No test plugin available!");
+        try {
+            String formatNames[] = ImageIO.getReaderFormatNames();
+            String readerMimeTypes[] = ImageIO.getReaderMIMETypes();
+            if (!isPresent(dummyformatNames, formatNames) ||
+                !isPresent(dummymimeType, readerMimeTypes)) {
+                throw new RuntimeException("No test plugin available!");
+            }
+        } catch (ServiceConfigurationError sce) {
+            System.out.println("Expected ServiceConfigurationError \n" + sce);
+            System.out.println("Test Passed");
```

Comment:
Previously, `ServiceConfigurationError` wasn't caught and, as far as I understand, wasn't expected; if it were thrown, the test would fail. Why is the logic different now?

Reply:
When SM is removed, test fails with java.util.ServiceConfigurationError: javax.imageio.spi.ImageReaderSpi:Provider BadReaderPluginSpi not found. This is the expected behavior without SM. 

When SM is present this error is swallowed because otherwise a erroneous plugin could cause a VM-wide problem. Now that SM is removed, the test (IIOPluginTest.java)
has been updated to expect ServiceConfigurationError.

AFAIK, there isn't another test which checks this code path hence it has been repurposed and retained.

Reply:
Yes, perhaps there's some more general test for this in java.base, but there's nothing wrong with verifying it here, even though it is sort of backward from what the test previously verified.
Probably the test could always have tested both cases.